File: src/main/java/com/hellsfood/api/users/data/UserRepository.java
package com.hellsfood.api.users.data;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

public interface UserRepository extends JpaRepository<User, Long> {
	boolean existsByUserId(String userId);

	boolean existsByEmail(String email);

	boolean existsByName(String name);

	Optional<User> findByUserId(String userId);

	@Query("SELECT u.userId from User u where u.email=:email and u.delFlag is null")
	Optional<String> findActiveUserIdByEmail(String email);

	@Query("SELECT MAX(u.id) from User u")
	Long getLastUID();
}

File: src/main/kotlin/com/hellsfood/api/users/data/UserRepository.kt
package com.hellsfood.api.users.data

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import java.util.*

interface UserRepository : JpaRepository<User, Long> {
    fun findByUserId(userId: String):User
    fun findIdByUserId(userId: String):Long

    @Query("SELECT u.userId from User u where u.email=:email")
    fun findUserIdByEmail(email: String?): Optional<String?>?
}